lib/dpl/provider/surge.rb

```diff
@@ -0,0 +1,33 @@
+module DPL
+  class Provider
+    class Surge < Provider
+      npm_g 'surge'
+
+      def project
+        File.expand_path("./" + (options[:project] || '') )
+      end
+
+      def domain
+        options[:domain] || ''
+      end
+
+      def check_auth
+		raise Error, "Please add SURGE_TOKEN in Travis settings (get your token with 'surge token')" unless context.env['SURGE_TOKEN']
+        raise Error, "Please add SURGE_LOGIN in Travis settings (its your email)" unless context.env['SURGE_LOGIN']
+      end
+
+      def check_app
+      	raise Error, "Please set a valid project folder path in .travis.yml under deploy: project: myPath" unless File.directory?(project)
```

Comment:
I'd avoid using the postfix conditionals here, since it is very had to see them with such a long message.